### DISCUSSION TITLE
What happens when max_size is reached?

### DISCUSSION THREAD

**Discussion by Roadtrp** [online-ml/river#853] [Q&A] [answered]
For models such as the adaptive random forest, what happens when the max_size parameter is reached? Does the model continue adapting and resetting base trees when drift is detected, or does it simply stop learning altogether? Thanks!

**Comment by smastelini** [comment, marked as answer]
Hi @Roadtrp, only tree statistics are updated, but no splits are performed. The drift detectors keep working and might end up dropping some trees, which frees some memory for growing new trees. 

Memory control is tricky in forests, because only the size of trees is affected. The other components in the ensembles are unbounded.
